tests/unit_tests/test_unit.py

```
@@ -1929,446 +2096,342 @@ def test_beam_search_decode(tiny_config):
+        tokenizer=depthcharge.tokenizers.peptides.PeptideTokenizer(
+            residues=config.residues
+        ),
+    )
+
+    model.max_peptide_len = 0
+    mzs = ints = torch.zeros(1, 5, device=model.device)
+    precursors = torch.tensor(
+        [[469.25364, 2.0, 235.63410]], device=model.device
+    )
+
+    assert len(list(model.beam_search_decode(mzs, ints, precursors))[0]) == 0
```

Ruff (0.15.2)

[warning] Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)